src/main/java/org/asamk/signal/dbus/DbusSignalImpl.java

```diff
@@ -248,24 +247,134 @@ public boolean isRegistered() {
     }
 
     @Override
-    public void updateProfile(
-            final String name,
-            final String about,
-            final String aboutEmoji,
-            String avatarPath,
-            final boolean removeAvatar
-    ) {
+    public void updateProfile(final String name, final String about, final String aboutEmoji, String avatarPath,
+            final boolean removeAvatar) {
         try {
             if (avatarPath.isEmpty()) {
                 avatarPath = null;
             }
-            Optional<File> avatarFile = removeAvatar
-                    ? Optional.absent()
+            Optional<File> avatarFile = removeAvatar ? Optional.absent()
                     : avatarPath == null ? null : Optional.of(new File(avatarPath));
             m.setProfile(name, about, aboutEmoji, avatarFile);
         } catch (IOException e) {
             throw new Error.Failure(e.getMessage());
         }
     }
 
+    // Provide option to query a version string in order to react on potential
+    // future interface changes
+    @Override
+    public String version() {
+        return BaseConfig.PROJECT_VERSION;
+    }
+
+    // Create a unique list of Numbers from Identities and Contacts to really get
+    // all numbers the system knows
+    @Override
+    public List<String> listNumbers() {
+        List<String> numbers = new ArrayList<>();
+
+        for (IdentityInfo identity : m.getIdentities()) {
+            String number = identity.getAddress().getNumber().orNull();
+            if (number != null) {
+                if (numbers.indexOf(number) == -1) {
+                    numbers.add(number);
+                }
+            }
+        }
+
+        var contacts = m.getContacts();
+        for (var c : contacts) {
+            if (numbers.indexOf(c.number) == -1) {
+                numbers.add(c.number);
+            }
+        }
+
+        return numbers;
+    }
+
+    @Override
+    public String getContactNumber(final String name) {
```

Comment:
This method is quite nondeterministic in case there are multiple profiles with the same name. Which is quite common as many people just use the first name.
How about making this method return a list of strings, so in case of duplicates multiple numbers can be returned?